Fix errors reported by TypeScript

<!--
Comments in any language are welcome. Please write in your native language.
Se aceptan comentarios en cualquier idioma. Por favor, escribe en tu lengua materna.
欢迎使用任何语言发表评论。请用您的母语书写。
모든 언어로 된 댓글을 환영합니다. 모국어로 작성해 주세요.
どの言語も歓迎します。あなたの母国語で書いてください。
-->

- Markuplint version: `4.9.1`
- Typescript version: `5.5.2`
- Node.js version: `20.14.0`
- OS: Ubuntu 22.04.4 LTS

## Describe the bug

TypeScript reports errors in Markuplint. I had similar issues : https://github.com/markuplint/markuplint/issues/1061, https://github.com/markuplint/markuplint/issues/1124, https://github.com/markuplint/markuplint/issues/1535.

## Code Example or Playground URL

- `package.json`

  ```json
  {
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
      "markuplint": "4.9.1",
      "typescript": "5.5.2"
    }
  }
  ```

- `index.ts`

  ```typescript
  import { MLEngine } from "markuplint";

  console.log(MLEngine);
  ```

## Steps To Reproduce

1. `npm install`
2. `npx tsc --lib esnext,dom --module nodenext index.ts`

## The current behavior

```
node_modules/@markuplint/ml-core/lib/ml-dom/node/attr.d.ts:21:14 - error TS2416: Property 'ownerElement' in type 'MLAttr<T, O>' is not assignable to the same property in base type 'Attr'.
  Type 'MLElement<T, O>' is missing the following properties from type 'Element': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

21     readonly ownerElement: MLElement<T, O>;
                ~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:26:9 - error TS2416: Property 'nextElementSibling' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type 'MLElement<T, O>' is missing the following properties from type 'Element': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

26     get nextElementSibling(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:35:9 - error TS2416: Property 'previousElementSibling' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

35     get previousElementSibling(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:39:5 - error TS2416: Property 'after' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 301 more.

39     after(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:44:5 - error TS2416: Property 'before' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

44     before(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:55:5 - error TS2416: Property 'replaceWith' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

55     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/comment.d.ts:5:22 - error TS2420: Class 'MLComment<T, O>' incorrectly implements interface 'Comment'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
              Types of parameters 'nodes' and 'nodes' are incompatible.
                Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

5 export declare class MLComment<T extends RuleConfigValue, O extends PlainData = undefined> extends MLCharacterData<T, O, MLASTComment> implements Comment {
                       ~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-fragment.d.ts:5:22 - error TS2420: Class 'MLDocumentFragment<T, O>' incorrectly implements interface 'DocumentFragment'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

5 export declare class MLDocumentFragment<T extends RuleConfigValue, O extends PlainData = undefined> extends MLParentNode<T, O, MLASTNode> implements DocumentFragment {
                       ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:25:5 - error TS2416: Property 'after' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

25     after(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:29:5 - error TS2416: Property 'before' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

29     before(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:37:5 - error TS2416: Property 'replaceWith' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

37     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:208:9 - error TS2416: Property 'doctype' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'DocumentType'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLElement<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLNode<any, any, MLASTNode>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

208     get doctype(): MLDocumentType<T, O> | null;
            ~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1453:5 - error TS2416: Property 'getElementById' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(elementId: string) => MLElement<T, O>' is not assignable to type '(elementId: string) => HTMLElement'.
    Type 'MLElement<T, O>' is missing the following properties from type 'HTMLElement': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

1453     getElementById(elementId: string): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1460:5 - error TS2416: Property 'getElementsByClassName' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.
      Type 'MLElement<T, O>' is missing the following properties from type 'Element': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

1460     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1460:66 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.

1460     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                      ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1474:5 - error TS2416: Property 'getElementsByTagName' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(qualifiedName: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '{ <K extends keyof HTMLElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<HTMLElementTagNameMap[K]>; <K extends keyof SVGElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<...>; <K extends keyof MathMLElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<...>; <K extends keyof HTMLElementDeprecatedTagNameMa...'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.

1474     getElementsByTagName(qualifiedName: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1474:67 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.

1474     getElementsByTagName(qualifiedName: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                       ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:386:9 - error TS2416: Property 'attributes' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type 'MLNamedNodeMap<T, O>' is not assignable to type 'NamedNodeMap'.
    The types of 'getNamedItem(...).ownerElement' are incompatible between these types.
      Type 'MLElement<T, O>' is missing the following properties from type 'Element': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

386     get attributes(): MLNamedNodeMap<T, O>;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:386:9 - error TS2416: Property 'attributes' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type 'MLNamedNodeMap<T, O>' is not assignable to type 'NamedNodeMap'.

386     get attributes(): MLNamedNodeMap<T, O>;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:575:9 - error TS2416: Property 'nextElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

575     get nextElementSibling(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:575:9 - error TS2416: Property 'nextElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

575     get nextElementSibling(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1426:9 - error TS2416: Property 'previousElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

1426     get previousElementSibling(): MLElement<T, O> | null;
             ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1426:9 - error TS2416: Property 'previousElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

1426     get previousElementSibling(): MLElement<T, O> | null;
             ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1549:5 - error TS2416: Property 'after' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLNode<any, any, MLASTNode>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

1549     after(...nodes: (string | MLNode<any, any>)[]): void;
         ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1549:5 - error TS2416: Property 'after' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLNode<any, any, MLASTNode>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

1549     after(...nodes: (string | MLNode<any, any>)[]): void;
         ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1581:5 - error TS2416: Property 'before' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

1581     before(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1581:5 - error TS2416: Property 'before' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

1581     before(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1661:5 - error TS2416: Property 'getAttributeNode' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Type 'MLAttr<T, O>' is not assignable to type 'Attr'.

1661     getAttributeNode(qualifiedName: string): MLAttr<T, O> | null;
         ~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1661:5 - error TS2416: Property 'getAttributeNode' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.

1661     getAttributeNode(qualifiedName: string): MLAttr<T, O> | null;
         ~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1703:5 - error TS2416: Property 'getElementsByClassName' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.
      Type 'MLElement<T, O>' is not assignable to type 'Element'.

1703     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1703:5 - error TS2416: Property 'getElementsByClassName' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.

1703     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1703:66 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.

1703     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                      ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1776:5 - error TS2416: Property 'insertAdjacentElement' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(where: InsertPosition, element: MLElement<T, O>) => MLElement<T, O>' is not assignable to type '(where: InsertPosition, element: Element) => Element'.
    Types of parameters 'element' and 'element' are incompatible.
      Type 'Element' is missing the following properties from type 'MLElement<T, O>': #private, closeTag, elementType, isForeignElement, and 182 more.

1776     insertAdjacentElement(where: InsertPosition, element: MLElement<T, O>): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1776:5 - error TS2416: Property 'insertAdjacentElement' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(where: InsertPosition, element: MLElement<T, O>) => MLElement<T, O>' is not assignable to type '(where: InsertPosition, element: Element) => Element'.

1776     insertAdjacentElement(where: InsertPosition, element: MLElement<T, O>): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1849:5 - error TS2416: Property 'replaceWith' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

1849     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1849:5 - error TS2416: Property 'replaceWith' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

1849     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:4:5 - error TS2416: Property 'getNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Call signature return types 'MLAttr<T, O>' and 'Attr' are incompatible.
      The types of 'ownerElement' are incompatible between these types.
        Type 'MLElement<T, O>' is missing the following properties from type 'Element': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

4     getNamedItem(qualifiedName: string): MLAttr<T, O> | null;
      ~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:12:5 - error TS2416: Property 'item' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(index: number) => MLAttr<T, O>' is not assignable to type '(index: number) => Attr'.
    Type 'MLAttr<T, O>' is not assignable to type 'Attr'.

12     item(index: number): MLAttr<T, O> | null;
       ~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:19:5 - error TS2416: Property 'removeNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Type 'MLAttr<T, O>' is not assignable to type 'Attr'.

19     removeNamedItem(qualifiedName: string): MLAttr<T, O>;
       ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:26:5 - error TS2416: Property 'removeNamedItemNS' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(namespace: string, localName: string) => MLAttr<T, O>' is not assignable to type '(namespace: string, localName: string) => Attr'.
    Type 'MLAttr<T, O>' is not assignable to type 'Attr'.

26     removeNamedItemNS(namespace: string | null, localName: string): MLAttr<T, O>;
       ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:33:5 - error TS2416: Property 'setNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(attr: MLAttr<T, O>) => MLAttr<T, O>' is not assignable to type '(attr: Attr) => Attr'.
    Types of parameters 'attr' and 'attr' are incompatible.
      Type 'Attr' is missing the following properties from type 'MLAttr<T, O>': #private, endQuote, equal, isDuplicatable, and 37 more.

33     setNamedItem(attr: MLAttr<T, O>): MLAttr<T, O> | null;
       ~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:40:5 - error TS2416: Property 'setNamedItemNS' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(attr: MLAttr<T, O>) => MLAttr<T, O>' is not assignable to type '(attr: Attr) => Attr'.
    Types of parameters 'attr' and 'attr' are incompatible.
      Type 'Attr' is not assignable to type 'MLAttr<T, O>'.

40     setNamedItemNS(attr: MLAttr<T, O>): MLAttr<T, O> | null;
       ~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:135:9 - error TS2416: Property 'childNodes' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
    Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
      Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
        Types of property 'after' are incompatible.
          Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

135     get childNodes(): NodeListOf<MLChildNode<T, O>>;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:135:34 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'Node'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.

135     get childNodes(): NodeListOf<MLChildNode<T, O>>;
                                     ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:143:9 - error TS2416: Property 'firstChild' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.

143     get firstChild(): MLChildNode<T, O> | null;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:159:9 - error TS2416: Property 'lastChild' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.

159     get lastChild(): MLChildNode<T, O> | null;
            ~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:170:9 - error TS2416: Property 'nextSibling' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.

170     get nextSibling(): MLChildNode<T, O> | null;
            ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:211:9 - error TS2416: Property 'parentElement' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLElement<T, O>' is missing the following properties from type 'HTMLElement': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

211     get parentElement(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:259:9 - error TS2416: Property 'parentNode' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLElement<T, O> | MLDocument<any, any> | MLDocumentFragment<any, any>' is not assignable to type 'ParentNode'.
    Type 'MLElement<T, O>' is not assignable to type 'ParentNode'.
      The types returned by 'children.namedItem(...)' are incompatible between these types.
        Type 'MLElement<T, O>' is missing the following properties from type 'Element': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

259     get parentNode(): MLDocument<any, any> | MLDocumentFragment<any, any> | MLElement<T, O> | null;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:274:9 - error TS2416: Property 'previousSibling' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.

274     get previousSibling(): MLChildNode<T, O> | null;
            ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:373:41 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.

373     conditionalChildNodes(): NodeListOf<MLChildNode<T, O>>[];
                                            ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:380:5 - error TS2416: Property 'contains' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type '(other: MLNode<T, O, MLASTNode>) => boolean' is not assignable to type '(other: Node) => boolean'.
    Types of parameters 'other' and 'other' are incompatible.
      Type 'Node' is missing the following properties from type 'MLNode<T, O, MLASTNode>': #private, MARKUPLINT_PREPROCESSOR_BLOCK, isFragment, rules, and 24 more.

380     contains(other: MLNode<T, O> | null): boolean;
        ~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:401:37 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.

401     getPureChildNodes(): NodeListOf<MLChildNode<T, O>>;
                                        ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:409:5 - error TS2416: Property 'getRootNode' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type '(options?: GetRootNodeOptions) => MLNode<T, O, MLASTNode>' is not assignable to type '(options?: GetRootNodeOptions) => Node'.
    Call signature return types 'MLNode<T, O, MLASTNode>' and 'Node' are incompatible.
      The types of 'childNodes' are incompatible between these types.
        Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.

409     getRootNode(options?: GetRootNodeOptions): MLNode<T, O>;
        ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:509:43 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.

509     resetChildren(childNodes?: NodeListOf<MLChildNode<T, O>>): void;
                                              ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20:9 - error TS2416: Property 'children' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollection'.
    The types returned by 'namedItem(...)' are incompatible between these types.
      Type 'MLElement<T, O>' is missing the following properties from type 'Element': setHTMLUnsafe, ariaBrailleLabel, ariaBrailleRoleDescription

20     get children(): HTMLCollectionOf<MLElement<T, O>>;
           ~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20:38 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.

20     get children(): HTMLCollectionOf<MLElement<T, O>>;
                                        ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:25:9 - error TS2416: Property 'firstElementChild' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

25     get firstElementChild(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:30:9 - error TS2416: Property 'lastElementChild' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

30     get lastElementChild(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:56:53 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Node'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLElement<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLNode<any, any, MLASTNode>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
              Types of parameters 'nodes' and 'nodes' are incompatible.
                Type 'string | Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
                  Type 'Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
                    Type 'Node' is missing the following properties from type 'MLNode<any, any, MLASTNode>': #private, MARKUPLINT_PREPROCESSOR_BLOCK, isFragment, rules, and 24 more.

56     querySelectorAll(selectors: string): NodeListOf<MLElement<T, O>>;
                                                       ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/text.d.ts:5:22 - error TS2420: Class 'MLText<T, O>' incorrectly implements interface 'Text'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

5 export declare class MLText<T extends RuleConfigValue, O extends PlainData = undefined> extends MLCharacterData<T, O, MLASTText> implements Text {
                       ~~~~~~


Found 61 errors in 11 files.

Errors  Files
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/attr.d.ts:21
     5  node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:26
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/comment.d.ts:5
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/document-fragment.d.ts:5
     3  node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:25
     6  node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:208
    19  node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:386
     6  node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:4
    13  node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:135
     5  node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/text.d.ts:5
```

## The expected behavior

No errors.

--- a/package.json
+++ b/package.json
@@ -97,7 +97,7 @@
 		"typedoc": "0.25.13",
 		"typedoc-plugin-mdn-links": "3.1.30",
 		"typedoc-plugin-resolve-crossmodule-references": "0.3.3",
-		"typescript": "5.4.5"
+		"typescript": "5.5.2"
 	},
 	"resolutions": {
 		"lerna/**/string-width": "4",

--- a/packages/@markuplint/create-rule/package.json
+++ b/packages/@markuplint/create-rule/package.json
@@ -30,7 +30,7 @@
 		"glob": "10.4.2",
 		"prettier": "3.2.5",
 		"ts-node": "10.9.2",
-		"typescript": "5.4.5"
+		"typescript": "5.5.2"
 	},
 	"devDependencies": {
 		"@types/fs-extra": "11.0.4",

--- a/packages/@markuplint/i18n/src/translator.ts
+++ b/packages/@markuplint/i18n/src/translator.ts
@@ -59,7 +59,7 @@ export function translator(localeSet?: LocaleSet): Translator {
 				if (Number.isNaN(num)) {
 					return $0;
 				}
-				const keyword = keywords[num] == null ? '' : toString(keywords[num]!, localeSet?.locale);
+				const keyword = keywords[num] == null ? '' : toString(keywords[num], localeSet?.locale);
 				// No translate
 				if (noTranslateIndex.has(number)) {
 					return keyword;

--- a/packages/@markuplint/ml-config/src/merge-config.ts
+++ b/packages/@markuplint/ml-config/src/merge-config.ts
@@ -227,7 +227,7 @@ function getName(item: any, comparePropName: string) {
 		return item;
 	}
 	if (typeof item === 'object' && item && comparePropName in item && typeof item[comparePropName] === 'string') {
-		return item[comparePropName] as string;
+		return item[comparePropName];
 	}
 	return null;
 }

--- a/packages/@markuplint/ml-core/src/ml-dom/helper/get-indent.ts
+++ b/packages/@markuplint/ml-core/src/ml-dom/helper/get-indent.ts
@@ -109,7 +109,7 @@ class MLDOMIndentation {
 		const lines = node.raw.split(/\r?\n/);
 		const index = this.line - line;
 		if (lines[index] != null) {
-			lines[index] = lines[index]!.replace(current, this.#fixed);
+			lines[index] = lines[index].replace(current, this.#fixed);
 		}
 		node.fix(lines.join('\n'));
 	}

--- a/packages/@markuplint/ml-core/src/ml-dom/node/element.ts
+++ b/packages/@markuplint/ml-core/src/ml-dom/node/element.ts
@@ -175,6 +175,28 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
+	get ariaBrailleLabel(): string | null {
+		throw new UnexpectedCallError('Not supported "ariaBrailleLabel" property');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @deprecated
+	 * @unsupported
+	 * @implements DOM API: `Element`
+	 */
+	get ariaBrailleRoleDescription(): string | null {
+		throw new UnexpectedCallError('Not supported "ariaBrailleRoleDescription" property');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @deprecated
+	 * @unsupported
+	 * @implements DOM API: `Element`
+	 */
 	get ariaBusy(): string | null {
 		throw new UnexpectedCallError('Not supported "ariaBusy" property');
 	}
@@ -3768,6 +3790,17 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 		attr: Attr,
 	): Attr | null {
 		throw new UnexpectedCallError('Not supported "setAttributeNodeNS" method');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @unsupported
+	 * @implements DOM API: `Element`
+	 * @see https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#dom-element-sethtmlunsafe
+	 */
+	setHTMLUnsafe(html: string): void {
+		throw new UnexpectedCallError('Not supported "setHTMLUnsafe" method');
 	}
 
 	/**

--- a/packages/@markuplint/pretenders/package.json
+++ b/packages/@markuplint/pretenders/package.json
@@ -31,6 +31,6 @@
 		"@markuplint/parser-utils": "4.6.4",
 		"glob": "10.4.2",
 		"meow": "13.2.0",
-		"typescript": "5.4.5"
+		"typescript": "5.5.2"
 	}
 }

--- a/packages/@markuplint/rules/src/table-row-column-alignment/grid.ts
+++ b/packages/@markuplint/rules/src/table-row-column-alignment/grid.ts
@@ -196,13 +196,13 @@ function createGrid(
 
 				if (matrix[rowNum]?.[colNum] === '↓' && !spanStart && colSpan > 1) {
 					// Overlap
-					matrix[rowNum]![colNum] = 'x';
+					matrix[rowNum][colNum] = 'x';
 				} else {
-					matrix[rowNum]![colNum] = spanStart ? '◎' : addCol > 0 ? '↓' : colSpan > 1 ? '→' : '●';
+					matrix[rowNum][colNum] = spanStart ? '◎' : addCol > 0 ? '↓' : colSpan > 1 ? '→' : '●';
 				}
 
 				if (addCol === 1) {
-					matrix[rowNum]!.push('●');
+					matrix[rowNum].push('●');
 				}
 
 				colNum += 1;

--- a/vscode/package.json
+++ b/vscode/package.json
@@ -126,7 +126,7 @@
 		"@vscode/test-electron": "2.4.0",
 		"@vscode/vsce": "2.29.0",
 		"esbuild": "0.21.5",
-		"typescript": "5.4.5"
+		"typescript": "5.5.2"
 	},
 	"dependencies": {
 		"@markuplint/esm-adapter": "4.5.0",

